docs/source-2.0/java/client/configuration.rst

```diff
@@ -0,0 +1,506 @@
+
+A number of auth schemes are provided by default in the client-http package. These include:
+
+*  :ref:`httpBearerAuth <httpBearerAuth-trait>` - Supports HTTP Bearer Authentication as defined in RFC 6750.
+*  :ref:`httpApiKeyAuth <httpApiKeyAuth-trait>` - Supports HTTP authentication using an API key sent in a header or query string parameter.
+*  :ref:`httpBasicAuth <httpBasicAuth-trait>` - Indicates that a service supports HTTP Basic Authentication as defined in RFC 2617.
+
+Add the ``client-http`` package as a dependency of your project to make these auth schemes available in your service.
+
+Worked Example: Adding HTTP API Key Authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Consider a Smithy modeled API for a service, ``ExampleService``. We would like to enable users of our generated SDK
+to authenticate to the API using an API key sent via an ``x-api-key`` HTTP header.
+Smithy Java already provides an ``httpApiKeyAuth`` Auth Scheme that we can use to allow
+this API Key authentication.
+
+Before we can add any Auth Scheme implementations to our generated client we must first add the associated
+:ref:`@httpApiKeyAuth <httpApiKeyAuth-trait>` scheme trait to our service model.
+
+.. code-block:: smithy
+    :caption: model.smithy
+
+    namespace com.example
+
+    @httpApiKeyAuth(name: "X-Api-Key", in: "header") // <- Add auth scheme trait
+    service ExampleService {
+        version: "2025-05-05"
+        // ...
+    }
+
+Authentication schemes are effectively part of your services interface and so (outside of testing)
+SHOULD always be modeled in your Smithy model using a trait. See the auth definition trait for more information
+on how to define a custom auth scheme in your Smithy model.
+
+Now that we have added our auth trait to the Smithy model we need to add a corresponding AuthScheme implementation
+to our client’s dependencies. The client-http package provides an ``HttpApiKeyAuthScheme`` implementation corresponding
+to the @httpApiKeyAuth trait.
+
+.. code-block:: kotlin
+    :caption: build.gradle.kts
+
+    dependencies {
+        // Add the HTTP auth scheme to the classpath
+        implementation("software.amazon.smithy.java:client-http:__smithy_java_version__")
+        // ...
+    }
+
+The ``HttpApiKeyAuthScheme`` class implements the ``AuthSchemeFactory`` service provider, making the auth scheme
+discoverable by the client codegen plugin.
+
+The built-in Smithy Java HTTP auth schemes all require one or more compatible ``IdentityResolver``
+to be register with the client. This resolver will handle actually fetching the clients credentials.
+The ``HttpApiKeyAuthScheme`` scheme needs an identity resolver that returns an ``ApiKeyIdentity``.
+For testing purposes we will provide a static resolver as follows:
+
+.. code-block:: java
+
+    var client = ExampleService.builder()
+            .addIdentityResolver(
+                IdentityResolver.of(new ApiKeyIdentity.create("example-api-key")
```

Comment:
```suggestion
                IdentityResolver.of(new ApiKeyIdentity.create("example-api-key"))
```

Reply:
Something in this block sets off at least GH's syntax highlighting, maybe it's this missing paren.